Lib/fontTools/designspaceLib/__init__.py

```diff
@@ -735,6 +735,21 @@ def read(self):
         self.readInstances()
         self.readLib()
 
+    def validateAttributes(self, element, attributeTypeMap):
+        for attributeName, attributeType in attributeTypeMap.items():
+            attrib = element.attrib.get(attributeName)
+            if attrib is None:
+                raise DesignSpaceDocumentError(
+                    "<%s> element missing required attribute '%s'"
+                    % (element.tag, attributeName))
+            try:
+                attributeType(attrib)
+            except Exception as e:
```

Comment:
i think you should only catch TypeError and ValueError here, not blind Exception